clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -423,9 +423,9 @@ synchronized int numAssignedPartitions() {
     }
 
     // Visible for testing
-    public synchronized List<TopicPartition> fetchablePartitions(Predicate<TopicPartition> isAvailable) {
+    // This method returns its results in a list provided as it is on hot path and causes considerable pressure on gc otherwise
```

Comment:
Can you add JavaDoc explaining that the result is stored in the final parameter?

Also, if we are not allocating a new List here, I don't see why we should return anything from this function.